src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+package mate.academy.dao;
+
+import mate.academy.ConnectionUtil;
+import mate.academy.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.services.Book;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book save(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection
+                     .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to insert at leas one row, but inserted 0 rows.");
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Book get(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+        PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                Integer price = resultSet.getObject("price", Integer.class);
+
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+
+                return book;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create a connection to the DB", e);
```

Comment:
The exception message 'Can not create a connection to the DB' might be misleading here. Consider changing it to something more specific, like 'Can not retrieve book with id: ' + id.